### ISSUE TITLE
Using show.bind on mdc-toolbar results in error

### ISSUE THREAD

**Issue by djensen47** [Ullfis/aurelia-mdc-bridge#31]
I tried to use `show.bind` on mdc-toolbar and got an error.

The toolbar:
```html
<mdc-toolbar class="mdc-elevation--z2" fixed="true" show.bind="true"></mdc-toolbar>
```

The error:
```
ERROR [app-router] TypeError: Cannot read property 'remove' of undefined
    at Animator.removeClass (http://localhost:9000/scripts/vendor-bundle.js:19227:25)
    at Show.valueChanged (http://localhost:9000/scripts/vendor-bundle.js:25773:23)
    at Show.bind (http://localhost:9000/scripts/vendor-bundle.js:25780:12)
    at Controller.bind (http://localhost:9000/scripts/vendor-bundle.js:22567:24)
    at View.bind (http://localhost:9000/scripts/vendor-bundle.js:20565:24)
    at Controller.bind (http://localhost:9000/scripts/vendor-bundle.js:22559:19)
    at Controller.automate (http://localhost:9000/scripts/vendor-bundle.js:22504:12)
    at ready (http://localhost:9000/scripts/vendor-bundle.js:26849:40)
    at RouterView.swap (http://localhost:9000/scripts/vendor-bundle.js:26878:14)
    at http://localhost:9000/scripts/vendor-bundle.js:17389:29
    at Array.forEach (<anonymous>)
    at http://localhost:9000/scripts/vendor-bundle.js:17388:20
From previous event:
    at NavigationInstruction._commitChanges (http://localhost:9000/scripts/vendor-bundle.js:17387:33)
    at CommitChangesStep.run (http://localhost:9000/scripts/vendor-bundle.js:17220:36)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18349:14)
    at inspect (http://localhost:9000/scripts/vendor-bundle.js:18326:16)
    at http://localhost:9000/scripts/vendor-bundle.js:18342:20
    at processPotential (http://localhost:9000/scripts/vendor-bundle.js:18460:14)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18341:18)
    at processActivatable (http://localhost:9000/scripts/vendor-bundle.js:18352:12)
    at ActivateNextStep.run (http://localhost:9000/scripts/vendor-bundle.js:18238:14)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18268:14)
    at processDeactivatable (http://localhost:9000/scripts/vendor-bundle.js:18271:12)
    at DeactivatePreviousStep.run (http://localhost:9000/scripts/vendor-bundle.js:18226:14)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18349:14)
    at inspect (http://localhost:9000/scripts/vendor-bundle.js:18326:16)
    at http://localhost:9000/scripts/vendor-bundle.js:18342:20
From previous event:
    at processPotential (http://localhost:9000/scripts/vendor-bundle.js:18430:35)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18341:18)
    at processActivatable (http://localhost:9000/scripts/vendor-bundle.js:18352:12)
    at CanActivateNextStep.run (http://localhost:9000/scripts/vendor-bundle.js:18214:14)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
From previous event:
    at LoadRouteStep.run (http://localhost:9000/scripts/vendor-bundle.js:18490:68)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at iterate (http://localhost:9000/scripts/vendor-bundle.js:18268:14)
    at processDeactivatable (http://localhost:9000/scripts/vendor-bundle.js:18271:12)
    at CanDeactivatePreviousStep.run (http://localhost:9000/scripts/vendor-bundle.js:18202:14)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at http://localhost:9000/scripts/vendor-bundle.js:17654:16
From previous event:
    at BuildNavigationPlanStep.run (http://localhost:9000/scripts/vendor-bundle.js:17652:58)
    at next (http://localhost:9000/scripts/vendor-bundle.js:17189:20)
    at Pipeline.run (http://localhost:9000/scripts/vendor-bundle.js:17202:14)
    at http://localhost:9000/scripts/vendor-bundle.js:18801:25
From previous event:
    at AppRouter._dequeueInstruction (http://localhost:9000/scripts/vendor-bundle.js:18774:32)
    at AppRouter.activate (http://localhost:9000/scripts/vendor-bundle.js:18751:12)
    at http://localhost:9000/scripts/vendor-bundle.js:18729:23
    at <anonymous>
From previous event:
    at AppRouter.registerViewPort (http://localhost:9000/scripts/vendor-bundle.js:18728:16)
    at new RouterView (http://localhost:9000/scripts/vendor-bundle.js:26768:19)
    at Object.invokeWithDynamicDependencies (http://localhost:9000/scripts/vendor-bundle.js:12534:20)
    at InvocationHandler.invoke (http://localhost:9000/scripts/vendor-bundle.js:12509:168)
    at Container.invoke (http://localhost:9000/scripts/vendor-bundle.js:12780:25)
    at ProviderResolver.get (http://localhost:9000/scripts/vendor-bundle.js:21149:74)
    at Container.get (http://localhost:9000/scripts/vendor-bundle.js:12719:23)
    at Container.elementContainerGet [as get] (http://localhost:9000/scripts/vendor-bundle.js:21204:17)
    at HtmlBehaviorResource.create (http://localhost:9000/scripts/vendor-bundle.js:23181:58)
    at applyInstructions (http://localhost:9000/scripts/vendor-bundle.js:21306:33)
    at ViewFactory.create (http://localhost:9000/scripts/vendor-bundle.js:21527:9)
    at HtmlBehaviorResource.create (http://localhost:9000/scripts/vendor-bundle.js:23193:41)
    at http://localhost:9000/scripts/vendor-bundle.js:23679:18
From previous event:
    at CompositionEngine.createController (http://localhost:9000/scripts/vendor-bundle.js:23678:10)
    at CompositionEngine._createControllerAndSwap (http://localhost:9000/scripts/vendor-bundle.js:23640:19)
    at CompositionEngine.compose (http://localhost:9000/scripts/vendor-bundle.js:23726:21)
    at TemplatingEngine.compose (http://localhost:9000/scripts/vendor-bundle.js:23989:38)
    at Aurelia.setRoot (http://localhost:9000/scripts/vendor-bundle.js:13230:21)
    at http://localhost:9000/scripts/app-bundle.js:328:22
From previous event:
    at Object.configure (http://localhost:9000/scripts/app-bundle.js:327:35)
    at http://localhost:9000/scripts/vendor-bundle.js:12059:29
From previous event:
    at config (http://localhost:9000/scripts/vendor-bundle.js:12054:56)
    at http://localhost:9000/scripts/vendor-bundle.js:12090:14
From previous event:
    at bootstrap (http://localhost:9000/scripts/vendor-bundle.js:12089:26)
    at http://localhost:9000/scripts/vendor-bundle.js:12076:9
    at <anonymous>
From previous event:
    at run (http://localhost:9000/scripts/vendor-bundle.js:12071:61)
    at Object.<anonymous> (http://localhost:9000/scripts/vendor-bundle.js:12096:37)
    at Object.execCb (http://localhost:9000/scripts/vendor-bundle.js:5484:33)
    at Module.check (http://localhost:9000/scripts/vendor-bundle.js:4671:51)
    at Module.enable (http://localhost:9000/scripts/vendor-bundle.js:4964:22)
    at Object.enable (http://localhost:9000/scripts/vendor-bundle.js:5345:39)
    at Module.<anonymous> (http://localhost:9000/scripts/vendor-bundle.js:4949:33)
    at http://localhost:9000/scripts/vendor-bundle.js:3922:23
    at each (http://localhost:9000/scripts/vendor-bundle.js:3847:31)
    at Module.enable (http://localhost:9000/scripts/vendor-bundle.js:4901:17)
    at Module.init (http://localhost:9000/scripts/vendor-bundle.js:4576:26)
    at http://localhost:9000/scripts/vendor-bundle.js:5248:36
```

**Comment by Ullfis** [comment]
I drifted badly into containerless mode in the start to make the design work. Such errors when different functions can't access `element` is some of the side effect of this.

I removed the containerless decorator of the toolbar and compared doc's before and after. New version on its way :smile:

**Comment by djensen47** [comment]
Thanks!

Does using containerless cause bad problems? I've found a few situations where I have to use it because the mdc css isnt expecting an extra element. For instance an mdc-checkbox inside of a list-item or if I use a custom element to output an mdc-grid-cell.

**Comment by Ullfis** [comment]
I used containerless for almost all components in the bridge. Like you say; the css isnt expecting extra element. I started to get some trouble like the one in this issue. It was hard to debug. And attributes like `class` needed to be handled.

I ended up in testing and removing containerless wherever I could. I also left some. Maybe time to check if some of these also can be removed.

External css style problems is one good reason to use containerless decorator.

These items still use containerless:
- grid-cell
- simple-menu
- tab
- toolbar-row
- toolbar-section
- toolbar-title

.. and has not been tested without.. yet.
